python/lbann/launcher/batch_script.py

```diff
@@ -0,0 +1,168 @@
+import os
+import os.path
+import subprocess
+
+class BatchScript:
+    """Utility class to write batch job scripts.
+
+    This class manages a non-interactive script file that can be
+    submitted as a batch job to an HPC job scheduler. A script is made
+    up of two parts: the header configures the job and the body
+    contains the actual commands to be executed.
+
+    This particular class is not fully implemented. Derived classes
+    for specific job schedulers should implement
+    `add_parallel_command` and `submit`, maintaining the same API.
+
+    """
+
+    def __init__(self,
+                 script_file=None,
+                 work_dir=os.getcwd(),
+                 interpreter='/bin/bash'):
+        """Construct batch script manager.
+
+        Args:
+            script_file (str): Script file.
+            work_dir (str, optional): Working directory
+                (default: current working directory).
+            interpreter (str, optional): Script interpreter
+                (default: /bin/bash).
+
+        """
+
+        # Lines in script are stored as lists of strings
+        self.header = []
+        self.body = []
+
+        # Construct file paths
+        self.work_dir = os.path.realpath(work_dir)
+        self.script_file = script_file
+        if not self.script_file:
+            self.script_file = os.path.join(self.work_dir, 'batch.sh')
+        self.script_file = os.path.realpath(self.script_file)
+        self.out_log_file = os.path.join(self.work_dir, 'out.log')
+        self.err_log_file = os.path.join(self.work_dir, 'err.log')
+
+        # Shebang line
+        if interpreter:
+            self.header.append('#!{}'.format(interpreter))
```

Comment:
It might be slightly more clean to use `add_header_line`.

Reply:
Done.